### PR TITLE
fix: remove deprecations from login form

### LINKED ISSUE #4055: Breaking Changes since v1.1.0: Login form component (#3140)
- [x] **Replace all constructors with:**
```
constructor(
    private auth: AuthService,
    private globalMessageService: GlobalMessageService,
    private fb: FormBuilder,
    private authRedirectService: AuthRedirectService,
    private winRef: WindowRef
  ) {}
```

- [x] **Replace if statement:**
```
	if (this.winRef && this.winRef.nativeWindow) { ... }
```
with
```
	if (this.winRef.nativeWindow) { ... }
```
NOTE: remove the first check for `this.winRef`

### PR DESCRIPTION
**Pull request by kacperknapik**
Closes #4055

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts [hunk 1]
```diff
@@ -37,22 +37,15 @@ export class LoginFormComponent implements OnInit, OnDestroy {
...
+    if (this.checkoutConfigService.isGuestCheckout()) {
+      this.loginAsGuest = this.activatedRoute?.snapshot?.queryParams?.['forced'];
     }
 
-    // TODO(issue:#4055) Deprecated since 1.1.0
-    if (this.winRef && this.winRef.nativeWindow) {
-      const routeState =
-        this.winRef.nativeWindow.history &&
-        this.winRef.nativeWindow.history.state;
+    const routeState = this.winRef?.nativeWindow?.history?.state;
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts]: Could you change that to prefilledEmail and move the last param to the chained properties?

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/login-form.component.migration.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  ANGULAR_FORMS,
+  AUTH_REDIRECT_SERVICE,
+  WINDOW_REF,
+  ACTIVATED_ROUTE,
+  ANGULAR_ROUTER,
+  CHECKOUT_CONFIG_SERVICE,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const LOGIN_FORM_COMPONENT_MIGRATION: ConstructorDeprecation = {
```
Comment (marlass) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/login-form.component.migration.ts:17]: Add this migration to `CONSTRUCTOR_DEPRECATION_DATA` array

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/login-form.component.migration.ts (hunk 2 above)
Comment (kacperknapik) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/data/login-form.component.migration.ts:17]: done

File: projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts (hunk 1 above)
Comment (kacperknapik) [reply projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts]: done

File: docs/migration/2_0.md [hunk 3]
```diff
@@ -129,6 +129,7 @@ Support for functions dropped:
 - `CartVoucherService` now requires new parameter `ActiveCartService`. This service needs to be provided for `CartVoucherService`.
 - `CartCouponComponent` no longer uses `CartService`. This service usage was replaced with corresponding methods from `ActiveCartService`. Also `CustomerCouponService` and `FeatureConfigService` are now required parameters. These services needs to be provided for `CartCouponComponent`.
 - `CartItemListComponent` no longer uses `CartService`. This service usage was replaced with corresponding methods from `ActiveCartService.` Also `SelectiveCartService` and `FeatureConfigService` are now required parameters. These services needs to be provided for `CartItemListComponent`.
+- `LoginFormComponent` requires new parameters now: `winRef`, `activatedRoute` and `checkoutConfigService`.
```
Comment (marlass) [inline docs/migration/2_0.md]:
```suggestion
- `LoginFormComponent` requires new parameters now: `WindowRef`, `ActivatedRoute` and `CheckoutConfigService`.
```

**Review by marlass — CHANGES_REQUESTED** [review]

File: docs/migration/2_0.md (hunk 3 above)
Comment (kacperknapik) [reply docs/migration/2_0.md]: done

**Review by marlass — APPROVED** [review]

**Pull request merged**